cloudflare/resource_cloudflare_healthcheck_test.go

```diff
@@ -0,0 +1,178 @@
+package cloudflare
+
+import (
+	"fmt"
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccCloudflareHealthcheckTCPExists(t *testing.T) {
+	t.Parallel()
+	zoneID := os.Getenv("CLOUDFLARE_ZONE_ID")
+
+	rnd := generateRandomResourceName()
+	name := fmt.Sprintf("cloudflare_healthcheck.%s", rnd)
+	var healthcheck cloudflare.Healthcheck
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckCloudflareHealthcheckTCP(zoneID, "tcp-test-healthcheck", rnd),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareHealthcheckExists(name, zoneID, &healthcheck),
+					resource.TestCheckResourceAttr(name, "description", ""),
+					resource.TestCheckResourceAttr(name, "port", "80"),
+					resource.TestCheckResourceAttr(name, "method", "connection_established"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccCloudflareHealthcheckTCPUpdate(t *testing.T) {
+	t.Parallel()
+	zoneID := os.Getenv("CLOUDFLARE_ZONE_ID")
+
+	rnd := generateRandomResourceName()
+	name := fmt.Sprintf("cloudflare_healthcheck.%s", rnd)
+	var healthcheck cloudflare.Healthcheck
+	var initialID string
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckCloudflareHealthcheckTCP(zoneID, "tcp-test-healthcheck-update", rnd),
```

Comment:
This is a good example of the second part I mentioned in https://github.com/terraform-providers/terraform-provider-cloudflare/pull/680/files#r423320742. We could use it here to have a guaranteed unique name and then confirm it in the config by appending `-updated` where needed.

Reply:
Updated here as well